ci/buildspec.yml

```diff
@@ -0,0 +1,31 @@
+version: 0.2
+env:
+  variables:
+    CONTEXT_LOCATION:
+    COMMIT_ID:
+    PR_NUMBER:
+    PATTERN_NAME:
+phases:
+  install:
+    runtime-versions:
+      nodejs: 18
+    commands:
+      - n 18.17.0
+      - |
+        if [ ! -z "${COMMIT_ID}" ]; then 
+          git fetch origin pull/${PR_NUMBER}/head:pr
+          git checkout main
+          # git merge needs user details, but we don't push anything, so the contents are unimportant
+          git -c "user.name=CI Bot" -c "user.email=dev@null" merge --no-edit ${COMMIT_ID}
+        fi
+      - npm i
+      - make build
+  pre_build:
+    commands:
+      - | 
+        [ -z "$CONTEXT_LOCATION" ] || aws s3 cp $CONTEXT_LOCATION .
+  build:
+    commands:
+      - make pattern "${PATTERN_NAME#/do-e2e-test }" "deploy --verbose --all --require-approval never"
+#    finally:
```

Comment:
Lets remove comments.

Reply:
I kept if we decide we need to clean up always